### ISSUE TITLE
When generating a sound over a stretched clip using a Nyquist generator, it produces wrong result

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#5371]
Discovered while testing #5350

#### STR:

1. Download & open [the project](https://www.dropbox.com/scl/fi/hquqrh7b3umy8uk5t8t5w/Nyquist-generator-bug.aup3?rlkey=yic310cn1bms1xavlokpo4358&dl=0)
2. Select an interval of appr. 0.5s
3. `Generate > Pluck` with the default length of 1.0s
4. Observe

#### macOS - correct result:

<img width="450" alt="Screenshot 2023-10-11 at 12 39 32" src="https://github.com/audacity/audacity/assets/70063970/27ac95fe-e6b9-4bd9-be4b-0894a5340c9d">

####  The generated pluck is shorter (~0.5s) and there's an unnecessary gap of silence of ~0.5s:

<img width="450" alt="Screenshot 2023-10-11 at 12 39 15" src="https://github.com/audacity/audacity/assets/70063970/30470416-b15e-40e1-82f1-45644dd27c9b">

**Comment by dozzzzer** [comment]
Let's agree to throw an error if the duration of the generated sound is longer than the selection.
@LWinterberg please provide a copy of the error message.

After showing such an error message, ideally, the User should be thrown back to the generator dialog, rather than to the project window. @saintmatthieu

**Comment by LWinterberg** [comment]
![image](https://github.com/audacity/audacity/assets/87814144/b0722507-02d0-4e2f-b28b-7f6748140a33)
this is the one currently used when trying to generate something that runs into some other clip. Probably could be made better as: 

> There's not enough room to generate audio of this length.

**Comment by saintmatthieu** [comment]
To reduce the risk of users facing this issue, we could easily change the Nyquist pluck generator dialog to use the selection length as the default length. All other syntheses I know of do that, and I do suppose it'd also be a UX improvement - @LWinterberg would that be okay ?

**Comment by LWinterberg** [comment]
I don't oppose that idea.

**Comment by saintmatthieu** [comment]
@dozzzzer could you try again on recent master? I can't reproduce it (on Windows)

**Comment by dozzzzer** [comment]
@saintmatthieu yes, it's still reproducible. However, it may take a few attempts to be able to replicate the bug.
For this, a simple stereo tone shrunk by e.g. 7% would be sufficient.


https://github.com/audacity/audacity/assets/70063970/e3f8fe6b-645b-40ec-bc71-6cac945b88a6
